ui/src/assets/worker/web/log.js

```diff
@@ -21,21 +21,25 @@ onmessage = function (e) {
 };
 
 function loadLog (user, session, api) {
-    if (user && api) {
+    loop(2, function () {
         var url = '/project/' + key + '/application/' + appName +
-            '/pipeline/' + pipName +'/build/' + buildNumber +
+            '/pipeline/' + pipName + '/build/' + buildNumber +
             '/action/' + jobID + '/step/' + stepOrder + '/log';
         if (envName !== '') {
             url += '?envName=' + envName;
         }
-        postMessage(httpCall(url, api, user, session));
-        setInterval(function () {
-            var stepLogs = httpCall(url, api, user, session);
-            postMessage(stepLogs);
-            var jsonLogs = JSON.parse(stepLogs);
+
+        var xhr = httpCall(url, api, user, session);
+        if (xhr.status >= 400) {
+            return true;
+        }
+        if (xhr.status === 200 && xhr.responseText !== null) {
```

Comment:
if xhr.responseText is undefined it will enter in the condition because you make a strict equals with null. Is it the right behavior ?